### PR TITLE
Addition for allowing ulimit overrides with the 'sudo' command on Ubuntu

### PR DESCRIPTION
**Pull request by arifcse019**
Addition for allowing ulimit overrides with the 'sudo' command on Ubuntu

### CODE REVIEW

**Comment by arifcse019** [comment]
Can you please review it? @bmhatfield @kisoku

File: recipes/default.rb [hunk 1]
```diff
@@ -22,6 +22,10 @@
     template "/etc/pam.d/su" do
       cookbook ulimit['pam_su_template_cookbook']
     end
+    cookbook_file "/etc/pam.d/sudo" do
```
Comment (bmhatfield) [inline recipes/default.rb]: Please add a newline between these :-)

File: recipes/default.rb [hunk 2]
```diff
@@ -22,6 +22,10 @@
...
+      source "sudo"
```
Comment (bmhatfield) [inline recipes/default.rb]: Please make `source` and `cookbook` optional/attribute driven here, so that if someone wants to do this differently, they can.

**Comment by bmhatfield** [comment]
I'd like to be *very* conservative about modifying `pam.d` on folks' machines, for what I hope are obvious reasons.

Can you expand on the rationale for this a bit more? Should it be optional? Is this, like `su`, really needed in the vast bulk of cases? Or is it less-used? Just looking for context here.

**Comment by arifcse019** [comment]
I tried to run a bats test to see if the file handle limit for an user is set correctly and the test command is this "sudo -u <theUser> -i ulimit -S -n". 

In ubuntu, after converging (i.e. changing file handle limit for deploy), depending on how you login, ulimit was returning different value. Using su was returning the changed value (4096) but using sudo was returning the default (1024). This was not a problem in centos. 

This Stack Overflow question has similarity with the problem I just described: http://stackoverflow.com/questions/11291307/different-values-returned-by-ulimit-r

**Comment by bmhatfield** [comment]
Alright, thank you for the information.

After reviewing my systems, I can see that only one line in `/etc/pam.d/sudo` has changed. The addition makes sense.

Please address the two pieces of feedback that I left directly on the changes, and I'll be happy to merge.

**Comment by arifcse019** [comment]
Great will do shortly

**Comment by arifcse019** [comment]
@bmhatfield  fixed those two

File: attributes/default.rb [hunk 3]
```diff
@@ -1,3 +1,5 @@
 default['ulimit']['pam_su_template_cookbook'] = nil
 default['ulimit']['users'] = Mash.new
 default['ulimit']['security_limits_directory'] = '/etc/security/limits.d'
+default['ulimit']['ulimit_overriding_sudo_file_name'] = 'sudo'
+default['ulimit'][''ulimit_overriding_sudo_file_cookbook''] = nil
```
Comment (bmhatfield) [inline attributes/default.rb]: Came here to merge this in, and noticed this accidental quoting issue - mind fixing it up real quick?

Thanks!

**Comment by arifcse019** [comment]
fixed the duplicate quote. Sorry about that

**Comment by arifcse019** [comment]
@bmhatfield How can we see the changes in the chef supermarket version of this cookbook? I downloaded the cookbook from supermarket but did not see the changes added.

**Pull request merged**